composer/trainer/trainer.py

```diff
@@ -1795,39 +1795,30 @@ def _train_microbatch(self, use_grad_scaling: bool, current_batch_size: int, tot
             with get_precision_context(self.state.precision):
                 self.state.loss = self._original_model.loss(self.state.outputs, self.state.batch)
 
-            # We always want to scale loss by the grad_accum before the backwards pass and
-            # also for sake of metrics. Complicating matters, the DeepSpeed engine does its
-            # own scaling when we call `.backward`, but this isn't in place so we still need
-            # to scale for sake of metrics after the `.backward` call.
-
-            # Loss is added to losses with clone to not scale the loss for the step printout
-            # Likely need to look into the performance impact
-            if not self.deepspeed_enabled:
-                for loss in ensure_tuple(self.state.loss):
-                    loss.mul_(microbatch_num_samples / current_batch_size)
-                    total_loss += loss.detach().clone()
-
             assert self.state.loss is not None
             self.engine.run_event(Event.AFTER_LOSS)
 
             # backward
             self.engine.run_event(Event.BEFORE_BACKWARD)
 
+            # Sum individual losses
+            microbatch_loss = self._device.tensor_to_device(torch.zeros(size=(1,)))
+            for loss in ensure_tuple(self.state.loss):
+                microbatch_loss += loss
```

Comment:
I always get a bit uneasy with operations like `+=` with autograd... could we do something like 
```python
microbatch_loss = torch.stack(ensure_tuple(self.state.loss)).sum()
```

Reply:
Sounds good to me!

Reply:
Hmm, this would allow `self.state.loss` to contain sequences of non-scalar tensors. Do you think this would be a desirable behavior or would it be best to check that each tensor is a scalar to avoid bugs?

Reply:
As an alternative, you can also do the inplace operator:

```suggestion
            microbatch_loss = self._device.tensor_to_device(torch.zeros(size=(1,)))
            for loss in ensure_tuple(self.state.loss):
                microbatch_loss.add_(loss)
```

My preference is for whatever is more generalizable